contracts/strategies/RFPStrategy.sol

```diff
@@ -0,0 +1,278 @@
+            (recipientId, recipientAddress, proposalBid, metadata) =
+                abi.decode(_data, (address, address, uint256, Metadata));
+
+            if (!_isIdentityManager(recipientId, _sender)) {
+                revert UNAUTHORIZED();
+            }
+        } else {
+            (recipientAddress, proposalBid, metadata) = abi.decode(_data, (address, uint256, Metadata));
+            recipientId = _sender;
+        }
+
+        // check if proposal bid is less than max bid
+        if (proposalBid > maxBid) {
+            revert EXCEEDING_MAX_BID();
+        } else if (proposalBid == 0) {
+            proposalBid = maxBid;
+        }
+
+        Recipient memory recipient = Recipient({
+            recipientAddress: recipientAddress,
+            proposalBid: proposalBid,
+            recipientStatus: RecipientStatus.Pending
+        });
+
+        recipients[recipientId] = recipient;
+        _recipientIds.push(recipientId);
+
+        emit Registered(recipientId, _data, _sender);
+    }
+
+    /// @notice Select recipient for RFP allocation
+    /// @param _data The data to be decoded
+    /// @param _sender The sender of the allocation
+    function _allocate(bytes memory _data, address _sender) internal override nonReentrant onlyPoolManager(_sender) {
+        if (acceptedRecipientId != address(0)) {
+            revert RECIPIENT_ALREADY_ACCEPTED();
+        }
+
+        acceptedRecipientId = abi.decode(_data, (address));
+
+        uint256 _recipientsCounter = _recipientIds.length;
+        for (uint256 i = 0; i < _recipientsCounter;) {
+            // update status of all other recipients to rejected
+            address recipientId = _recipientIds[i];
+            if (recipientId != acceptedRecipientId) {
+                recipients[recipientId].recipientStatus = RecipientStatus.Rejected;
+            }
+
+            unchecked {
+                i++;
+            }
+        }
+
+        // update status of acceptedRecipientId to accepted
+        if (acceptedRecipientId != address(0)) {
+            Recipient storage recipient = recipients[acceptedRecipientId];
+            recipient.recipientStatus = RecipientStatus.Accepted;
+
+            IAllo.Pool memory pool = allo.getPool(poolId);
+
+            emit Allocated(acceptedRecipientId, recipient.proposalBid, pool.token, _sender);
+        }
+    }
+
+    function _distribute(address[] memory, bytes memory, address _sender) internal override onlyPoolManager(_sender) {
+        if (acceptedRecipientId == address(0)) {
+            revert NO_ACCEPTED_RECIPIENT();
+        }
+
+        if (upcomingMilestone > milestones.length) {
+            revert INVALID_MILESTONE();
+        }
+
+        IAllo.Pool memory pool = allo.getPool(poolId);
+        Milestone storage milestone = milestones[upcomingMilestone];
+        Recipient memory recipient = recipients[acceptedRecipientId];
+
+        uint256 amount = recipient.proposalBid * milestone.amountPercentage / 1e18;
```

Comment:
we need to track the total percentage of all milestones somewhere. Else we would be able to distribute more than 100% of the `proposalBid` in total.
(for example if you have 5 milestones with 25% each)

Reply:
Maybe we loop all milestones when setMilestones is invoked and we ensure total is 100% ?